python-threatexchange/threatexchange/api.py

```diff
@@ -601,3 +601,101 @@ def get_threat_descriptors_from_indicator(
         response = self.get_json_from_url(url)
 
         return response["descriptors"]["data"]
+
+
+class NonThreatExchangeAPI:
+    _BASE_URL = "https://nciiapi.azure-api.net/api"
+
+    def __init__(
+        self,
+        x_functions_key: str,
+        ocp_apim_subscription_key: str,
+        start_timestamp: int,
+        *,
+        endpoint_override: t.Optional[str] = None,
+        page_size: int = 1000,
+    ) -> None:
+        self.x_functions_key = x_functions_key
+        self.ocp_apim_subscription_key = ocp_apim_subscription_key
+        self.start_timestamp = start_timestamp
+        self.page_size = page_size
+        self._base_url = endpoint_override or self._BASE_URL
+
+    def get_json_from_url(
+        self, url, params=None, *, json_obj_hook: t.Callable = None, headers=None
+    ):
+        """
+        Perform an HTTP GET request, and return the JSON response payload.
+        Same timeouts and retry strategy as `_get_session` above.
+        """
+        with self._get_session() as session:
+            response = requests.get(url, params=params or {}, headers=headers or {})
+            response.raise_for_status()
+            return response.json(object_hook=json_obj_hook)
+
+    def _get_session(self):
```

Comment:
a lot methods look like complete copy of the ones above docstring and all. Can we consolidate?